app/src/main/java/de/danoeh/antennapod/fragment/BaseHomeFragment.java

```diff
@@ -0,0 +1,128 @@
+import android.support.v4.app.FragmentPagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.MainActivity;
+
+//This fragment is the base fragment of the Home Page. The HomePageFragment and CategoriesFragment will be
+//loaded in the tabs contained in the BaseHomeFragment
+public class BaseHomeFragment extends Fragment{
+
+    public static final String TAG = "BaseHomeFragment";
+
+    private static final String LAST_TAB = "tab_position";
+
+    private SharedPreferences prefs;
+
+    //positions of the tabs
+    public static final int POS_FEATURED = 0;
+    public static final int POS_CATEGORIES = 1;
+    public static final int TOTAL = 2;
+
+    private TabLayout tabLayout;
+    private ViewPager viewPager;
+
+    private MainActivity act;
+
+    public void setAct(MainActivity mainActivity){
+        this.act = mainActivity;
+    }
+    public BaseHomeFragment(){
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setHasOptionsMenu(true);
+    }
+
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        setHasOptionsMenu(true);
+        this.setAct((MainActivity) getActivity());
+        act.getSupportActionBar().setTitle(R.string.homepage_label);
+
+        View rootView = inflater.inflate(R.layout.base_home_fragment, container, false);
+        viewPager = (ViewPager)rootView.findViewById(R.id.home_viewpager);
+        viewPager.setAdapter(new HomePagerAdapter(getChildFragmentManager(), getResources()));
+
+        // Give the tabs layout the viewpager
+        tabLayout = (TabLayout) rootView.findViewById(R.id.home_tabs);
+        tabLayout.setupWithViewPager(viewPager);
+
+        return rootView;
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        // save the tab selection
+        prefs = getActivity().getSharedPreferences(TAG, Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = prefs.edit();
+        editor.putInt(LAST_TAB, tabLayout.getSelectedTabPosition());
+        editor.apply();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        // restore our last position
+        prefs = getActivity().getSharedPreferences(TAG, Context.MODE_PRIVATE);
+        int lastPosition = prefs.getInt(LAST_TAB, 0);
+        viewPager.setCurrentItem(lastPosition);
+    }
+
+
+    //HomePagerAdapter is a static inner class of BaseHomeFragment
+    public static class HomePagerAdapter extends FragmentPagerAdapter {
+
+        private final Resources resources; //it is required in order to have the constructor
+        private Fragment[] fragments = {
+                new HomePageFragment(),
```

Comment:
If you are going to keep this fragment for the featured tab, please rename this fragment so that it is more intuitive to others.

Reply:
I will change the names